src/image_tools/completions.py

```diff
@@ -0,0 +1,102 @@
+from .args import build_bake_argparser
+
+
+def nushell_completion(completions):
+    # List of options to exclude
+    excluded_options = {"--product"}
+
+    # Collect completion lines
+    # One line for each argument we support in bake
+    completion_lines = []
+    for completion in completions:
+        if completion["long"] not in excluded_options:
+            # Build the baseline
+            if completion["short"]:
+                line = f"  {completion['long']} ({completion['short']})"
+            else:
+                line = f"  {completion['long']}"
+
+            # Add ": string" if it takes a parameter
+            if completion["takes_param"]:
+                line += ": string"
+
+            # Add help text as a comment if available
+            if completion["help"]:
+                line += f" # {completion['help']}"
+
+            # Append the line to the list
+            completion_lines.append(line)
+    completion_lines = "\n".join(completion_lines)
+
+    script_template = f"""
+def list_products_completions [] {{
+  let output = (bake --list-products | from json)
+  let products = $output | columns
+  return $products
+}}
+
+def list_versions_completions [product] {{
+  let output = (bake --list-products | from json)
+  let versions = $output | get $product | each {{|version| $"($product)=($version)"}}
+  return $versions
+}}
+
+export extern "bake" [
+{completion_lines}
+  --product: string@"nu-complete product" # Product to build images for. For example 'druid' or 'druid=28.0.1' to build a specific version.
+]
+
+def "nu-complete product" [context: string] {{
+  # This function will not currently work properly should one full product be a prefix of another product name
+
+  # context will be something like "bake --product <something>"
+  # So, after splitting it up we'll have
+  # - Row 0: "bake"
+  # - Row 1: "--product"
+  # - Row 2: "<something>" (e.g. empty or "hb" or "hbase" or "hbase=2.6.0" or similar
```

Comment:
I'll need to read up on this in more detail, but hardcoding "product comes first" seems.. wonky.

Reply:
Looking through this more, it looks like there's actually no reliable way to inspect the current argument. The completion can take the current position-in-the-string as an argument, but there is no (seemingly) reliable way to parse into a sequence of arbitrary arguments. Maybe it would be better to just not be clever and always include both variants.

Reply:
I'm not sure I understand your comment, sorry. Could you try to rephrase the issue for me?

Reply:
`split row ' '` (which is equivalent to `split words`) will split `"a 'b c' d"` into `["a", "'b", "c'", "d"]`, but running it as a command would parse it as `["a", "b c", "d"]`. That makes it hard to reliably answer things like "what is the current value of --product?". That also doesn't cover other syntaxes like (`--product=zookeeper`).

Reply:
I still don't understand I'm afraid....

Reply:
I would do something like:

```nushell
def products [] {
    bake --list-products | from json | transpose name versions
}

def products_and_versions [] {
    products | each {|product|
        [
            $product.name,
            ($product.versions | each {|version| $"($product.name)=($version)" })
        ] | flatten
    } | flatten
}

export extern "bake" [
  # ...snipped out
  --product (-p): string@products_and_versions # Product to build images for. For example 'druid' or 'druid=28.0.1' to build a specific version.
]
```

That way, you always get completions for both variants (`kafka` and `kafka=3.7.1`), but it still seems to prefer the wildcard version (`kafka`).

Reply:
I had that at the beginning but that makes the list hard(er) to read. I prefer the 2-step completion even if it's not perfect.

Reply:
Let's go with it as is for now then, and revisit later. I guess at worst I'll just end up running a local fork or something.

Reply:
Thank you. I'm happy to have this improved later.
This is totally optional and the resulting file can easily be overridden anyway.